### PR TITLE
PDP-1281: Sim swap feature added

### PR DESCRIPTION
**Pull request by danibishop**
* This feature counts the number of users within a time widow that share the same device id
* Such device id is extracted from a target user
* The idea behind is to detect SIM swapping. This means that the same device is used with different phone numbers.
* Structure has been simplified, avoiding pythonic syntax in favour of easy to read for non-Python native developers.

### CODE REVIEW

File: lib/count_sim_swaps/main.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import os
+from pngme.api import AsyncClient
+
+
+async def get_count_of_sim_swaps(
+    api_client: AsyncClient,
+    user_uuid: str,
+    utc_timestamp: datetime,
+    semi_window_days: int,
+) -> int:
```
Comment (frankie-ug) [inline lib/count_sim_swaps/main.py:12]: Can we make the return type Optional since we can return `None`

File: lib/count_sim_swaps/main.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+from datetime import datetime, timezone
+import os
+from pngme.api import AsyncClient
+
+
+async def get_count_of_sim_swaps(
+    api_client: AsyncClient,
+    user_uuid: str,
+    utc_timestamp: datetime,
+    semi_window_days: int,
```
Comment (frankie-ug) [inline lib/count_sim_swaps/main.py:11]: How about we use `utc_endtime` so we maintain the same method signatures as other features?

**Review by frankie-ug — CHANGES_REQUESTED** [review]
Nice work! Just two minor comments

File: lib/count_sim_swaps/main.py (hunk 2 above)
Comment (danibishop) [reply lib/count_sim_swaps/main.py:11]: The issue is that this is not an end time, it is a "window central time". It is not that I am in love with `uct_timestamp` but end time does not fit. Another suggestion?

File: lib/count_sim_swaps/main.py (hunk 1 above)
Comment (danibishop) [reply lib/count_sim_swaps/main.py:12]: You are totally right. How is that the linter is not complaining?!

File: lib/count_sim_swaps/main.py (hunk 2 above)
Comment (frankie-ug) [reply lib/count_sim_swaps/main.py:12]: Gotcha 👍

**Review by frankie-ug — APPROVED** [review]

**Pull request merged**